### PR TITLE
Jq transformer

### PR DESCRIPTION
**Pull request by tibbycat**

### CODE REVIEW

**Comment by tlblessing** [comment]
fyi @eskibars 

Direct review link: https://deploy-preview-2411--kongdocs.netlify.app/hub/tibbycat/jq-transformer/

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+#########################
+# PLUGIN-ONLY SETTINGS below this line
+# If your plugin is an extension of the core platform, ALL of the following
+# lines must be completed.
+# If NOT defined as a 'plugin' in line 32, delete all lines up to '# BEGIN MARKDOWN CONTENT'
+
+params: # Metadata about your plugin
+  name: jq-transformer
+  service_id:
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:83]:
```suggestion
  service_id: true 
```

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    # Boolean - whether this plugin can be applied to a Service.
+    # Affects generation of examples and config table.
+  consumer_id:
+    # Boolean - whether this plugin can be applied to a Consumer.
+    # Affects generation of examples and config table.
+  route_id:
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:89]:
```suggestion
  route_id: true
```

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    # whether this plugin can be applied to a Route.
+    # Affects generation of examples and config table.
+  protocols:
+    # List of protocols this plugin is compatible with, in array format.
+    # Valid values: "http", "https", "tcp", "tls"
+    ["http", "https"]
+  dbless_compatible:
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:96]: IS this plugin compatible with db-less mode? k8? Please see tabs for OSS and Enterprise here https://docs.konghq.com/hub/plugins/compatibility/

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    # Boolean - enables or disables all autogenerated examples (cURL, YAML, and
+    # Kubernetes). Default is `true`; set to `false` to disable all
+    # autogenerated examples.
+
+  config: # Configuration settings for your plugin
+    - name: Request
+      required: no
+      default: empty string
+      value_in_examples:
+        # If the field is to appear in examples, this is the value to use.
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:119]: Can you provide examples for request & response strings?

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+categories:
+  - transformations
+type: plugin
+desc: Extends kong with a wrapper for the amazing ./jq JSON processor
+description: |
+  Trivially transforms your request or response body with the https://stedolan.github.io/jq/ processor if the JSON payloads are slightly mis-aligned, without modifying code in your end service.
+  * Transform the request & response to simplify the interface to your end service (eg Confluent's kafka REST proxy)
+  * Remove sensitive(/private) information from your response payload.
+  * Reduce your response payload size.
+  NB: Content-Type header will probably affect your results. (application/.*json is probably the safest)
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:13]:
```suggestion

  <div class="alert alert-warning">
    <strong>Note: </strong> The Content-Type header will probably affect your results (`application/.*json` is probably the safest).
  </div>
```

**Review by tlblessing — CHANGES_REQUESTED** [review]
Suggested edits and requests for more information. We could add the install instructions and examples from the readme at the end of the file.

Marking as Do not merge for now until all information received and the plugin is approved for merging by @eskibars.

File: app/_hub/tibbycat/jq-transformer/index.md (hunk 3 above)
Comment (tibbycat) [reply app/_hub/tibbycat/jq-transformer/index.md:99]: I am unsure of this as I don’t use dB-less mode. However, I don’t explicitly interact with the dB. And commonly use decK to apply the plugin.

Comment (tibbycat) [reply app/_hub/tibbycat/jq-transformer/index.md:99]: Would it be acceptable to say ‘partially’ as this ‘could be’ but is ‘untested’?

Comment (tlblessing) [reply app/_hub/tibbycat/jq-transformer/index.md:99]: Probably not but I'll defer to @eskibars

File: app/_hub/tibbycat/jq-transformer/index.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    - name: Response
+      required: no
+      default: ""
+      value_in_examples:
+        {"custom":"Additional custom value for client"}+.
+      description:
+        ./jq expression to transform response body by
+  #  - name: # add additional setting blocks as needed, each demarcated by -
+  extra:
+    If you wish to remove sensitive data. For example {"name":"tibbycat","sex":"male"}, you could transform this by the expression {"name":.name}
```
Comment (tlblessing) [inline app/_hub/tibbycat/jq-transformer/index.md:134]:
```suggestion
    If you want to remove sensitive data such as {"name":"tibbycat","sex":"male"}, you could transform this using the expression {"name":.name}
```

**Review by tlblessing — CHANGES_REQUESTED** [review]
Another review comment.

**Comment by tlblessing** [comment]
Note to self: Clean out template instructions prior to merging.

**Comment by GayleNeumann** [comment]
@tibbycat
Thank you for the plugin submission for our Kong Hub. Unfortunately, we are not accepting new plugin entries at this time as we are in the process of revising our Plugin Hub and plugin submission criteria. We will have updates to our contributing guidelines soon and ask that you check back in a month or so. Again, thank you for your submission as we appreciate your efforts and contributions.  
Best, 
Gayle Neumann
Documentation Manager

**Pull request closed without merging**